source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.md

```diff
@@ -576,6 +578,87 @@ other threads from checking out [Connections](#connection) while establishing a
 Before a given [Connection](#connection) is returned from checkOut, it must be marked as "in use", and the pool's
 availableConnectionCount MUST be decremented.
 
+##### Draining of Pending Response (drivers that support CSOT)
+
+If an operation times out while awaiting a server response:
+
+- **Non-exhaust commands**: If the command was not sent with the `exhaustAllowed` OP_MSG bit flag, the driver MUST:
+    1. Put the connection into a "pending response" state.
+    2. Record the current time in a way that can be updated while draining of pending response.
+- **Exhaust commands**: If the command was sent with the `exhaustAllowed` bit, the driver MUST NOT transition the
+    connection to a "pending response" state and MUST instead close the connection.
+
+The next time connection in the "pending response" state is checked out, the driver MUST ensure that any remaining
+response data is drained and the connection is ready to send the next command.
+
+1. **Persist and update timestamp**: The connection must record the current time immediately after the original timeout.
+    This timestamp MUST be updated to the current time whenever any bytes are successfully read, received, or consumed
+    while draining of the pending response.
+2. **Aliveness check**: If driver uses synchronous read from the socket and the timestamp is older than 3 seconds, the
+    driver MUST perform aliveness check either performing a non-blocking read or using the minimal possible timeout to
+    check if at least one byte can be read/received. If at least one byte can be read the timestamp must be updated and
+    connection SHOULD be drained. If no bytes can be read, the connection MUST be closed.
+3. **User-provided timeout**: If a user-provided timeout is specified for the operation, the driver MUST use the minimum
+    of (a) the remaining time before the 3 second "pending response" window elapses and (b) the remaining of
+    user-provided timeout as the effective timeout for the draining.
+4. **Default timeout**: If no user-provided timeout is specified, the driver MUST use the minimum of (a) the remaining 3
+    second "pending response" window and (b) the `waitQueueTimeoutMS` (if not supported by the driver remaining of
+    `serverSelectionTimeoutMS` should be used instead) as the effective timeout for the draining.
```

Comment:
```suggestion
4. **Default timeout**: If no user-provided timeout is specified, the driver MUST use the minimum of (a) the remaining 3
    second "pending response" window and (b) remaining timeout for connection checkout.
```

regardless of whether CSOT is enabled or not or waitQueueTimeoutMS is supported, the timeout should always be `min(remaining connection checkout timeout, remaining static timeout)`.  With the existing phrasing, drivers would apply a full `waitQueueTimeoutMS` to every pending read attempt.